tests/capabilities/test-class-register-capabilities.php

```diff
@@ -18,6 +19,54 @@ public function test_register() {
 		$registered = $manager->get_capabilities();
 
 		$this->assertContains( 'wpseo_bulk_edit', $registered );
+		$this->assertContains( 'wpseo_edit_advanced_metadata', $registered );
 		$this->assertContains( 'wpseo_manage_options', $registered );
 	}
+
+	/**
+	 * Tests that capabilities are correctly unset if only super admins have access.
+	 *
+	 * @group ms-required
+	 *
+	 * @dataProvider data_filter_user_has_wpseo_manage_options_cap
+	 *
+	 * @covers WPSEO_Register_Capabilities::filter_user_has_wpseo_manage_options_cap()
+	 */
+	public function test_filter_user_has_wpseo_manage_options_cap( $role, $access, $expected_has_cap ) {
+		WPSEO_Options::get_instance();
+
+		$options           = get_site_option( 'wpseo_ms', array() );
+		$options['access'] = $access;
+		update_site_option( 'wpseo_ms', $options );
+
+		$register = new WPSEO_Register_Capabilities();
+		$register->register();
+
+		if ( $role === 'network_administrator' ) {
+			$user = self::factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+			grant_super_admin( $user->ID );
+		} else {
+			$user = self::factory()->user->create_and_get( array( 'role' => $role ) );
+		}
+
+		$allcaps = $register->filter_user_has_wpseo_manage_options_cap( $user->allcaps, array( 'wpseo_manage_options' ), array(), $user );
+
+		$has_cap = ! empty( $allcaps['wpseo_manage_options'] );
+		if ( $expected_has_cap ) {
+			$this->assertTrue( $has_cap );
+		} else {
+			$this->assertFalse( $has_cap );
+		}
+	}
+
+	public function data_filter_user_has_wpseo_manage_options_cap() {
```

Comment:
Can you describe the testing values? 

I've tried to do this in the premium plugin. There is no real good example or format for it. But every piece of explanation will help. 
![cursor](https://user-images.githubusercontent.com/325040/40353499-ef445f96-5db1-11e8-8a6d-5f6fc2d99a14.png)